Streetcode/Streetcode.BLL/Middleware/ApiRequestResponseMiddleware.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            var originalBodyStream = context.Response.Body;
+
+            var request = await GetRequestAsTextAsync(context.Request);
+
+            _loggerService.LogInformation(request);
+
+            await using var responseBody = new MemoryStream();
+            context.Response.Body = responseBody;
+
+            await next(context);
+
+            var response = await GetResponseAsTextAsync(context.Response);
+
+            var filteredResponse = GetResponseWithMaxLength(response, context.Request.Path);
+
+            _loggerService.LogInformation(filteredResponse);
+
+            await responseBody.CopyToAsync(originalBodyStream);
+        }
+
+        private async Task<string> GetRequestAsTextAsync(HttpRequest request)
+        {
+            var requestBody = request.Body;
+
+            request.EnableBuffering();
+
+            var buffer = new byte[Convert.ToInt32(request.ContentLength)];
+
+            await request.Body.ReadAsync(buffer, 0, buffer.Length);
+
+            var bodyAsText = Encoding.UTF8.GetString(buffer);
+
+            request.Body = requestBody;
+
+            return $"{request.Scheme} {request.Host} {request.Path} {request.QueryString} {bodyAsText}";
+        }
+
+        private async Task<string> GetResponseAsTextAsync(HttpResponse response)
+        {
+            response.Body.Seek(0, SeekOrigin.Begin);
+
+            var encoding = Encoding.UTF8;
+
+            var text = await new StreamReader(response.Body, encoding).ReadToEndAsync();
+
+            response.Body.Seek(0, SeekOrigin.Begin);
+
+            return text;
+        }
+
+        private string GetResponseWithMaxLength(string response, string requestPath)
+        {
+            try
+            {
+                var jsonObject = JToken.Parse(response);
+
+                if (jsonObject.Type == JTokenType.Array)
+                {
+                    var jsonArray = (JArray)jsonObject;
+
+                    foreach (var item in jsonArray)
+                    {
+                        TruncateProperties(item);
+                    }
+                }
+                else if (jsonObject.Type == JTokenType.Object)
+                {
+                    TruncateProperties(jsonObject);
+                }
+
+                return jsonObject.ToString();
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+        }
+
+        private void TruncateProperties(JToken token)
+        {
+            foreach (var property in token.Children<JProperty>().ToList())
+            {
+                if (!_loggerPropertiesExceptions.Contains(property.Name.ToString()))
+                {
+                    if (property.Value.Type == JTokenType.String && property.Value.ToString().Length > 100)
+                    {
+                        property.Value = new JValue(property.Value.ToString().Substring(0, 100));
+                    }
```

Comment:
Let's make this max length restriction configurable from appsettings.json along with the property name to ignore

Reply:
Please let me know how to do it, because for now I have this:
appsettings:
 "PropertiesToIgnore": [
      "id",
      "base64",
      "description"
    ]
code:
     public ApiRequestResponseMiddleware(ILoggerService loggerService, IConfiguration configuration)
        {
            _loggerService = loggerService;
            _configuration = configuration;
            _loggerPropertiesExceptions = _configuration.GetSection("Serilog:PropertiesToIgnore").Get<string[]>();
        }